### PR TITLE
Outliers sphinx documentation

### PR DESCRIPTION
**Pull request by kperrynrel**
## Description

This documentation covers the outliers module in PVAnalytics, and provides examples for the tukey, hampel, and z-score functions. It is based on https://github.com/pvlib/pvanalytics/issues/133.

*Thank you for your contribution! Please provide a brief description of the problem and the proposed solution or new feature (if not already fully described in a linked issue)*

## Checklist

The following items must be addressed before the code can be merged. 
Please don't hesitate to ask for help if you are unsure of how to accomplish any of the items. 
*You are free to remove any checklist items that do not apply or add additional items that are 
not on this list*

- [X] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings
- [x] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- [X] Non-API functions clearly documented with docstrings or comments as necessary
- [X] Pull request is nearly complete and ready for detailed review

### CODE REVIEW

File: pvanalytics/quality/outliers.py [hunk 1]
```diff
@@ -63,7 +63,8 @@ def zscore(data, zmax=1.5, nan_policy='raise'):
 
     """
     data = data.copy()
-    nan_mask = pd.Series([False] * len(data))
+    nan_mask = pd.Series([False] * len(data),
+                         index=data.index)
```
Comment (kanderso-nrel) [inline pvanalytics/quality/outliers.py:67]: Should this be considered a bugfix?  If so, it deserves a whatsnew note

**Review by kanderso-nrel — COMMENTED** [review]
I know the examples already on `main` follow this formula of showing the "right answer" as a basis of comparison for what the function actually returns, but it seems silly to me to have exactly the same thing shown in three different examples.  To me it seems like we could take them out without really losing anything.  @kperrynrel what do you think?

File: pvanalytics/quality/outliers.py (hunk 1 above)
Comment (kperrynrel) [reply pvanalytics/quality/outliers.py:67]: Ah, yes, good point. I'll add it to the whatsnew

File: docs/whatsnew/0.1.2.rst [hunk 2]
```diff
@@ -10,6 +10,8 @@ Enhancements
 Bug Fixes
 ~~~~~~~~~
 
+* Updated the outliers routine for z-score, where the NaN mask is assigned the time
+  series index (:issue:`133`, :pull:`138`)
```
Comment (kanderso-nrel) [inline docs/whatsnew/0.1.2.rst:14]:
```suggestion
  series index (:pull:`138`)
```

**Review by kanderso-nrel — APPROVED** [review]
LGTM!

File: docs/examples/tukey-outlier-detection.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# %%
+# Identifying and removing outliers from PV sensor time series
+# data allows for more accurate data analysis.
+# In this example, we demonstrate how to use
+# :py:func:`pvanalytics.quality.outliers.tukey` to identify and filter
+# out outliers in a time series.
+# We use a normalized time series example provided by the PV Fleets Initiative.
+# This example is adapted from the DuraMAT DataHub
+# clipping data set:
+# https://datahub.duramat.org/dataset/inverter-clipping-ml-training-set-real-data
```
Comment (cwhanse) [inline docs/examples/tukey-outlier-detection.py:18]: Move this text to the next comment paragraph, to keep the url close to the file name.

File: docs/examples/tukey-outlier-detection.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+from pvanalytics.quality.outliers import tukey
+import matplotlib.pyplot as plt
+import pandas as pd
+import pathlib
+
+# %%
+# First, we read in the ac_power_inv_7539_outliers example. Normalized AC power
+# is represented by the "value_normalized" column. There is a boolean column
+# "outlier" where inserted outliers are labeled as True, and all other values
+# are labeled as False.
```
Comment (cwhanse) [inline docs/examples/tukey-outlier-detection.py:30]: I understand what is being done but wonder if we're being a bit terse here.  The data contain normalized AC power (or energy?) at 15 minute intervals, where power is normalized by the inverter's AC capacity (?). For the purpose of illustrating outlier detection, we have overwritten a few lines of data with extreme values.  Is that correct?

**Review by cwhanse — APPROVED** [review]
@kperrynrel looks good, two comments which apply to all three examples.

File: docs/examples/tukey-outlier-detection.py (hunk 4 above)
Comment (kperrynrel) [reply docs/examples/tukey-outlier-detection.py:30]: @cwhanse that is correct. The data is actually min-max normalized and is in 15-minute intervals. I added a smattering of extreme outliers to illustrate the function performance when detecting outliers. I can make this more explicit in our documentation if that would help

Comment (kperrynrel) [reply docs/examples/tukey-outlier-detection.py]: @cwhanse I made some updates in the commenting per your recommendations, and made the outlier insertion commenting more explicit. Let me know if this works and I'll merge!

**Pull request merged**